File: 1518.WaterBottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int cnt = numBottles;
        while (numBottles >= numExchange) {
            cnt += numBottles / numExchange;
            numBottles = numBottles/numExchange + numBottles%numExchange;
        }
        return cnt;
    }
}

File: 1000-1999/1518.WaterBottles.go
func numWaterBottles(numBottles int, numExchange int) int {
	// O(1)
	return numBottles + (numBottles-1)/(numExchange-1)
	// O(logN)
	cnt := numBottles
	for numBottles >= numExchange {
		cnt += numBottles / numExchange
		numBottles = numBottles/numExchange + numBottles%numExchange
	}
	return cnt
}